File: Programming and Projection Methods/Lab5C#/Lab5/Lab5/repository/IBookingRepository.cs
using Lab5.domain;
using Lab5.utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab5.repository
{
    public interface IBookingRepository:IRepository<KeyValuePair<Ride, Client>,Booking>
    {
        IEnumerable<RBooking> findBookingsByDestDateHour(String destination, String date, String hour);
    }
}

File: Programming and Projection Methods/Lab7/Laborator7/CompanyPersistance/src/main/java/company/persistance/IBookingRepository.java
package company.persistance;

import company.model.Booking;
import company.model.Client;
import company.model.RBooking;
import company.model.Ride;
import javafx.util.Pair;

public interface IBookingRepository extends IRepository<Pair<Client, Ride>, Booking> {
    Iterable<RBooking> findBookingsByDestDateHour(String destination, String date, String hour);
}
